### DISCUSSION TITLE
Im having an issue testing a component that uses new "use" hook from react 18

### DISCUSSION THREAD

**Discussion by bmirandan** [vitest-dev/vitest#6421] [Q&A]
Hi! 

Here is the situation:

A ssr component providing a child component - client side a bunch of promises and consuming them by using the new brand "use" hook.

Similar to the example provided by react.

<img width="968" alt="Screenshot 2024-08-28 at 19 54 04" src="https://github.com/user-attachments/assets/ec11fef4-f58c-41ef-b18b-1ea9a717bb31">

When i run the test im getting " Error: Uncaught [TypeError: __vite_ssr_import_1__.use is not a function] " 

Is there any unsupported versions of react by vite? 
can be related with my logic? 
is there a known workaround? like for instance mock the function?

Have a great day!

**Comment by hi-ogawa** [comment]
I don't think `React.use` is available in stable react 18 as it's warned in the doc https://react.dev/reference/react/use
You need to use 19 which is available from rc, beta, etc... https://www.npmjs.com/package/react?activeTab=versions

Btw, I'm a little confused. This is a Vitest discussion, but is your question purely about Vite SSR?
Or maybe you're actually using Next.js for main app, but using Vitest for testing?
It would be great if you can share more context.

**Reply by bmirandan** [reply]
Yeah, sorry. As u said. 

Im using 
 * Next Js: 14.2.5 (app router) 
 * Vitest : 1.6.0 (as test runner)
 * React: 18.3.1

**Reply by hi-ogawa** [reply]
Next.js vendors React and internally uses latest React for your app, but Vitest of course has no idea about that.
You'll probably need to bump Next 15 rc and React 19 rc https://nextjs.org/blog/next-15-rc
